docs/source/loading.mdx

```diff
@@ -220,6 +220,22 @@ Load a list of Python dictionaries with [`~Dataset.from_list`]:
 >>> dataset = Dataset.from_list(my_list)
 ```
 
+### Python generator
+
+Create a dataset from a Python generator with [`~Dataset.from_generator`]:
+
+```py
+>>> from datasets import Dataset
+>>> def my_gen():
+...     yield {"a": 1}
+...     yield {"a": 2}
+...     yield {"a": 3}
```

Comment:
(nit) you can use a for loop instead, most users will probably use a for loop IMO